### PR TITLE
Context for docs

### PR DESCRIPTION
**Pull request by mariebeersangvig**
PR for improvements to Docs. Contains more examples, more context and more explanations and definitions.

### CODE REVIEW

File: best-practice.rst [hunk 1]
```diff
@@ -233,6 +300,407 @@ Datasets
...
 • name them the same as the pipe that produced it (the default and does not need to be specified)
 
+.. _best-practice-workflow:
+
+Workflow for transforming data in Sesam
+---------------------------------------
+
+Most Sesam projects will have a set flow that the data goes through.
+
+The data fed into Sesam through **input pipes** where namespaced identity is added in order to keep existing data model with joins intact. In addition **RDF type** for future filtering and classification is added in the input pipe.
```
Comment (TarjeiSkrede) [inline best-practice.rst:310]: "The data is fed into.."

Comment (TarjeiSkrede) [inline best-practice.rst:310]: "In addtion is a **RDF type** added in the input pipe for future filtering and classification."

File: best-practice.rst [hunk 2]
```diff
@@ -233,6 +300,407 @@ Datasets
...
+
+**Global pipes** merge data belonging together to generate **global datasets**. To be able to easily spot a global pipe, the following code can be added:
+
+::
+
+  "metadata": {
+    "global": true
+ }
+
+**Preparation pipes** is where **global datasets** are prepared for target systems. It is here most of logic is added. It could include enriching with more context from other datasets, structuring data into other formats, adding new fields and other transformations. The main purpose is to get data ready for target system.
```
Comment (TarjeiSkrede) [inline best-practice.rst:320]: "It is here most of the logic is added."

File: best-practice.rst [hunk 3]
```diff
@@ -233,6 +300,407 @@ Datasets
...
+
+**Output pipes** has no logic and basically sends data to endpoint.
```
Comment (TarjeiSkrede) [inline best-practice.rst:322]: "**Output pipes** basically sends data to an endpoint and should normally have no logic."

File: best-practice.rst [hunk 4]
```diff
@@ -327,9 +327,9 @@ The data fed into Sesam through **input pipes** where namespaced identity is add
     "global": true
  }
 
-**Preparation pipes** is where **global datasets** are prepared for target systems. It is here most of logic is added. It could include enriching with more context from other datasets, structuring data into other formats, adding new fields and other transformations. The main purpose is to get data ready for target system.
+**Preparation pipes** is where **global datasets** are prepared for target systems. It is here most of the logic is added. It could include enriching with more context from other datasets, structuring data into other formats, adding new fields and other transformations. The main purpose is to get data ready for the target system.
```
Comment (ashkanvahidishams) [inline best-practice.rst:330]: explanation for why we do the transforms here would be nice.

File: best-practice.rst [hunk 5]
```diff
@@ -327,9 +327,9 @@ The data fed into Sesam through **input pipes** where namespaced identity is add
...
 
-**Output pipes** has no logic and basically sends data to endpoint.
+**Output pipes** basically sends data to an endpoint and should normally have no logic.
```
Comment (ashkanvahidishams) [inline best-practice.rst:332]: same here.

File: best-practice.rst [hunk 6]
```diff
@@ -494,7 +494,7 @@ In the global pipe we want to add a metadata tag to show this is a pipe going in
 
   "metadata": {
     "global": true
- }
+ }dc
```
Comment (ashkanvahidishams) [inline best-practice.rst:497]: ?

File: getting-started.rst [hunk 7]
```diff
@@ -372,9 +450,29 @@ First we need to be able to assort the different people into two separate groups
...
 The third line is activated if the statement is true, and the fourth line if the statement is false.
 
-Another handy function is "return" which allows us to specify which values we want returned from source when doing hops.
+Another handy function is ``["return"]``  which allows us to specify which values we want returned from source when doing hops. Please see code example below
+
+Config:
+
+::
+
+  ["hops", {
```
Comment (ashkanvahidishams) [inline getting-started.rst:459]: kan hende, siden den mangler ytterste nivå som feks add, så blir det litt komplisert å forstå hvordan denne fungerer.

Comment (ashkanvahidishams) [reply getting-started.rst:459]: eller vil denne fungere i det heletatt som outputen litt ned beskriver?

File: getting-started.rst [hunk 8]
```diff
@@ -1178,20 +1389,22 @@ Inside your Demo_MicroserviceProject folder create a new file called Dockerfile
...
+Next step is to create the "requirements.txt" inside the "service" folder and paste the following text inside it:
 
 ::
 
  Flask==1.0.2
 
-Then create a python file, also in the "service" folder, named "DemoMicroservice.py" with the following code:
+ If you have a newer version of Flask, you put that in instead of 1.0.2.
+
+Final part is the actual program. FOr this we create a python file, also in the "service" folder, named "DemoMicroservice.py" with the following code:
```
Comment (ashkanvahidishams) [inline getting-started.rst:1407]: FOr

**Review by ashkanvahidishams — CHANGES_REQUESTED** [review]

**Review by ashkanvahidishams — APPROVED** [review]

**Pull request merged**